client/src/templates/Challenges/redux/completion-epic.js

```diff
@@ -181,11 +181,16 @@ export default function completionEpic(action$, state$) {
       const pathToNavigateTo = async () => {
         return await findPathToNavigateTo(nextChallengePath, superBlock);
       };
-
+      let result = false;
       return submitter(type, state).pipe(
+        tap(res => {
+          result = res.type !== 'app.updateFailed';
+        }),
         concat(closeChallengeModal),
-        filter(Boolean),
-        finalize(async () => navigate(await pathToNavigateTo()))
+
+        finalize(async () => {
+          result && navigate(await pathToNavigateTo());
+        })
```

Comment:
```suggestion
        tap(res => {
          if (res.type === baseActionTypes.submitComplete)
            navigate(pathToNavigateTo());
        })
```
Rather than using finalize, which will trigger in response to all events, we can navigate if we get the expected action.

Also, while I was digging into this I realised that `pathToNavigateTo` does not need to be async, so that should be dropped.  It makes life a little easier!

Finally, `baseActionTypes` needs importing from `'../../../redux/action-types'` so don't have to use the string.